internal/xds/translator/runner/runner.go

```diff
@@ -121,7 +121,11 @@ func (r *Runner) subscribeAndTranslate(sub <-chan watchable.Snapshot[string, *ir
 				result.EnvoyPatchPolicyStatuses = nil
 
 				// Publish
-				r.Xds.Store(key, result)
+				if err == nil || r.Config.EnvoyGateway.XDS.UpdateSnapshotOnError {
```

Comment:
my preference is to make this the default case and not expose a configuration (yet)
by making this change
```
if err == nil {
  r.Xds.Store(key, result)
}
```
the behaviour looks like the panic recovery case, where we dont push the config to the xds server which holds the last good state

this change may surface bugs that should have been caught in CEL or gateway-api layer, but hoping those get caught in staging and fixed in patch releases